packages/recomponents/src/components/r-select/r-select.vue

```diff
@@ -636,15 +661,21 @@
                 }
                 return options.slice(0, this.optionsLimit);
             },
+            hasContent() {
+                return (this.options.length !== 0 || this.showNoOptions)
+                    && (this.showNoResults || this.filteredOptions.length !== 0 || !this.search);
+            },
             hasLabel() {
                 return (this.label || '').trim() !== '';
             },
             classes() {
-                return this.isInvalid ? 'r-is-error' : '';
+                return {
+                    'r-is-error': this.isInvalid,
+                };
             },
             isInvalid() {
                 if (this.validate) {
-                    return this.validate.$invalid && this.validate.$dirty;
+                    return this.validate.$invalid && this.value.length !== 0;
```

Comment:
Shouldn't we be checking for `$error` instead of `$invalid`? 
https://vuelidate.js.org/#sub-v-values

Also what is the purpose of the length check? We already have an option for `allow empty`, and if the user wants to specify invalid-on-empty they should be adding that as a rule in `this.validate` no?

Reply:
I think this tag validation can be performed outside of RSelect component, added story in this PR: https://github.com/Rebilly/rebilly-recomponents/pull/146